include/internal/catch_tostring.h

```diff
@@ -174,18 +181,47 @@ namespace Detail {
     std::string rangeToString( InputIterator first, InputIterator last );
 }
 
-//template<typename T, typename Allocator>
-//struct StringMaker<std::vector<T, Allocator> > {
-//    static std::string convert( std::vector<T,Allocator> const& v ) {
-//        return Detail::rangeToString( v.begin(), v.end() );
-//    }
-//};
 
+// toString for containors
+// Sequence containors
+template<typename T, typename Allocator>
+std::string toString( std::vector<T,Allocator> const& c ) {
+    return Detail::rangeToString( c.begin(), c.end() );
+}
+template<typename T, typename Allocator>
+std::string toString( std::deque<T,Allocator> const& c ) {
+    return Detail::rangeToString( c.begin(), c.end() );
+}
+template<typename T, typename Allocator>
+std::string toString( std::list<T,Allocator> const& c ) {
+    return Detail::rangeToString( c.begin(), c.end() );
+}
+
+#if defined(CATCH_CPP11_OR_GREATER)
+template<typename T, std::size_t N>
+std::string toString( std::array<T,N> const& c ) {
+    return Detail::rangeToString( c.begin(), c.end() );
+}
```

Comment:
A generic container toString with enable_if begin/end-checks does not work?
Would also remove the need to include all the headers.

```cpp
template <typename T>
std::string toString(const T& c )
{
    return Detail::rangeToString(c.begin(), c.end());
}
```